### PR TITLE
`StrimziKafkaCluster` KRaft mode

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds support for KRaft within `StrimziKafkaCluster`. Currently, we are able to run in KRaft mode with the single-node instance.

### CODE REVIEW

File: .checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
         </module>
         <module name="CyclomaticComplexity">
             <!-- default is 10-->
-            <property name="max" value="11"/>
```
Comment (scholzj) [inline .checkstyle/checkstyle.xml:101]: If you need to override this for some method, you should use annotation and not change the rules for everything.

Comment (see-quick) [reply .checkstyle/checkstyle.xml]: Thanks. I simplify code a little bit and this is not needed anymore :)

**Review by scholzj — APPROVED** [review]
LGTM assuming it works 😉. At some poinbt we should also update the README. But that can be done in separate PR. Thanks.

**Review by jankalinic — APPROVED** [review]

**Comment by see-quick** [comment]
> LGTM assuming it works 😉. At some poinbt we should also update the README. But that can be done in separate PR. Thanks.

Yeah, I will update README in separate PR.

**Pull request merged**